src/linker/Linker/TypeMapInfo.cs

```diff
@@ -29,18 +29,79 @@
 using System.Collections.Generic;
 using Mono.Cecil;
 
-namespace Mono.Linker.Steps
+namespace Mono.Linker
 {
 
-	public class TypeMapStep : BaseStep
+	public class TypeMapInfo
 	{
+		readonly LinkContext _context;
+		AnnotationStore Annotations => _context.Annotations;
```

Comment:
Not a fan of this circular dependency. Could you move default_interface_implementations cache to this type to avoid that?